src/components/Card.js

```diff
@@ -1,41 +1,74 @@
+import { data } from "autoprefixer";
+
 export default class Card {
-  constructor(data, cardSelector, handleImageClick){
-    this.title = data.title;
-    this.url = data.url;
+  constructor(data, cardSelector, handleImageClick, handleDeleteClick, handleLikeClick, handleDislikeClick){
+    this.name = data.name;
+    this.link = data.link;
+    this._id = data._id;
+    this._isLiked = data.isLiked
     this._cardSelector = cardSelector;
     this._handleImageClick = handleImageClick;
+    this._handleDeleteClick = handleDeleteClick
+    this._handleLikeClick = handleLikeClick;
+    this._handleDislikeClick = handleDislikeClick;
   };
   
-  _delete(){
-    this._cardElement.remove();
+  getCardId(){
+    return this._id
+  }
+
+  delete(){
+    this._cardElement.remove()
   };
   
   _like(){
-    this._cardElement.querySelector('.card__heart').classList.toggle("card__heart_active");
+    this._cardElement.querySelector('.card__heart').classList.add("card__heart_active");
+    this._handleLikeClick(this.getCardId())
   };
 
+  _dislike(){
+    this._cardElement.querySelector('.card__heart').classList.remove('card__heart_active')
+    this._handleDislikeClick(this.getCardId())
+  }
+
   _setEventListeners(){
     this._cardImageElement = this._cardElement.querySelector('.card__image');
 
     this._cardElement.querySelector('.card__heart').addEventListener('click', ()=>{
-      this._like();
+      if (this._isLiked){
+        this._dislike(this)
+      } else {
+        this._like(this)
+      }
+      this._isLiked = !this._isLiked
```

Comment:
`      this._isLiked = !this._isLiked
`

You need to **toggle likes** only in block `then` after a successful response from the server because if there is an error the user will not know about that and will see the toggled heart color.